### PR TITLE
Fix Quantum Jobs Readme and Sample issues #19467 

### PR DESCRIPTION
**Pull request by v-hongli1**
Fix issue https://github.com/Azure/azure-sdk-for-java/issues/19467

@scottaddie , @joshfree and @jongio for notification.

### CODE REVIEW

**Comment by v-xuto** [comment]
@jongio - Who should we ping to review this PR?

**Review by JonathanGiles — APPROVED** [review]

**Comment by v-xuto** [comment]
@JonathanGiles  - Could you help us merge this PR?

File: sdk/quantum/azure-quantum-jobs/src/samples/java/com/azure/quantum/jobs/ReadmeSamples.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public void uploadInputData() {
         BlobClient blobClient = new BlobClientBuilder()
             .endpoint(inputDataUri)
             .buildClient();
-        String problemFilePath = FileSystems.getDefault().getPath("src/samples/resources/problem.json").toString();
+        String problemFilePath = FileSystems.getDefault().getPath("{uploadFilePath}").toString();
```
Comment (vxfield) [inline sdk/quantum/azure-quantum-jobs/src/samples/java/com/azure/quantum/jobs/ReadmeSamples.java:91]: The previous path points to an existing file in the repo which contains a working problem to be submitted and the customer can run the sample without doing any modification.
Why are we replacing it with `{uploadFilePath}`?

**Review by vxfield — CHANGES_REQUESTED** [review]

File: sdk/quantum/azure-quantum-jobs/src/samples/java/com/azure/quantum/jobs/ReadmeSamples.java (hunk 1 above)
Comment (JonathanGiles) [reply sdk/quantum/azure-quantum-jobs/src/samples/java/com/azure/quantum/jobs/ReadmeSamples.java:91]: When I reviewed this I was a little unsure too, but if I were to guess, the answer would be that this is a sample for showing in a readme file without all the usual Java verbosity around it (which [can be seen here](https://github.com/Azure/azure-sdk-for-java/blob/2027900ef496aa660e2cae9a72e5a05c13bde4cc/sdk/quantum/azure-quantum-jobs/README.md#upload-input-data)), and there was some desire to simplify the text to make it clear where user inputs must be edited. I'm personally ok with the previous text being retained - I see no need to make it more generic.

Comment (hongli750210) [reply sdk/quantum/azure-quantum-jobs/src/samples/java/com/azure/quantum/jobs/ReadmeSamples.java]: According to your comment, it has been corrected in the new version.

**Review by JonathanGiles — APPROVED** [review]

**Pull request merged**